docs/troubleshooting-guide/performance.md

```diff
@@ -0,0 +1,113 @@
+
+6. The database executes the query and returns the result to the back end.
+
+7. The back end forwards those results to the front end.
+
+8. The front end creates the HTML to display them in your browser.
+
+## Specific Problems
+
+Each of the steps described is a potential performance bottleneck. Before diving into specifics, you may want to read our article on [Metabase at scale][metabase-at-scale].
+
+### The Metabase instance is getting so much traffic that loading the HTML page is slow.
+
+This is extremely rare, since our front end is not very large and browsers cache our JavaScript, but it's easy to rule out. What is more common is the lack of HTTP/2 or the lack of available connections to the application database; we cover both topics in our article on [Metabase at scale][metabase-at-scale].
+
+**How to detect this:** Nothing appears on the page (not even the controls), or elements appear one by one.
+
+**How to fix this:**
+
+1.  Check that you are using HTTP/2 by looking at the network tab in your browser and checking the response headers.
+2.  Check the server logs for Metabase's application database.
+3.  Look at page load times in the browser console. If it's taking a long time to load our HTML, CSS, or JavaScript, check to see whether a proxy, firewall, or other network component is slowing things down.
+
+### Caching is disabled.
+
+By default caching is disabled so that we always re-run every question. However, if your data is only being updated every few seconds or minutes, you will improve performance by enabling caching. Note that ad hoc queries and exports are *not* cached, so if you're doing a lot of either, performance may suffer.
+
+**How to detect this:** Open the Admin Panel, go to "Settings", and look in the "Caching" tab to see whether caching is enabled or not.
+
+**How to fix this:** [This guide][admin-caching] explains how to change the minimum query duration (we cache anything that takes longer than that to run) and the maximum cache size for each query result. You may need to experiment with these values over several days to find the best balance.
+
+### The answer you want isn't cached.
+
+Cached values are stored in the application database, so they will still be there if Metabase restarts, provided that the cache duration is still valid. Each question (and any filter combination) is its own query, so if different users are viewing the same question with different filters, each will have to load once before it's cached. This is particularly obvious with [data sandboxing][data-sandboxing], because each user essentially has their own query for each question.
+
+**How to detect this:** Look at Metabase's logs, or in the server's logs, to see when it was last restarted.
+
+**How to fix this:** Patience is a virtue.
```

Comment:
While I agree, we should say that the query needs to run more than 2 times to Metabase can do an average of query run times to detect that a question should be cached

Reply:
A query is cached on first run - not on average - if it takes more time than the duration defined in the cache settings.
But the sections was rewritten, so the "How to detect this" and "How to fix this" needs an update.

Reply:
rewritten